python/cudf/cudf/core/column/string.py

```diff
@@ -143,33 +140,19 @@ def __init__(
             and dtype.kind == "U"
         ):
             dtype = CUDF_STRING_DTYPE
-        if len(children) > 1:
-            raise ValueError("StringColumn must have at most 1 offset column.")
-
-        if len(children) == 0 and size != 0:
-            # all nulls-column:
-            offsets = as_column(0, length=size + 1, dtype=SIZE_TYPE_DTYPE)
-
-            children = (offsets,)
 
         super().__init__(
-            data=data,
+            plc_column=plc_column,
             size=size,
             dtype=dtype,
-            mask=mask,
             offset=offset,
             null_count=null_count,
-            children=children,
+            exposed=exposed,
         )
 
         self._start_offset = None
         self._end_offset = None
 
-    def copy(self, deep: bool = True) -> Self:
```

Comment:
Since string columns follow the behavior of other columns now, do we need any new tests?